tls/ttls.c

```diff
@@ -2287,6 +2289,9 @@ ttls_recv(void *tls_data, unsigned char *buf, size_t len, unsigned int *read)
 		}
 	}
 
+	if (len == 0)
+		return T_POSTPONE;
+
```

Comment:
Not an issue of this patch, but later at `skip_record` label we have `WARN_ON_ONCE(delta > len)` statement. It's good to catch the issue and throw a warning, but more unpredictable situations may happen in that case. Better to close the connection in that case:
```
if (WARN_ON_ONCE(delta > len))
    return T_BLOCK;
```